llvm/lib/Target/AMDGPU/AMDGPU.td

```diff
@@ -119,6 +119,12 @@ def FeatureUnalignedDSAccess : SubtargetFeature<"unaligned-ds-access",
   "Hardware supports unaligned local and region loads and stores"
 >;
 
+def FeatureRequireNaturallyAlignedBufferAccess : SubtargetFeature<"require-naturally-aligned-buffer-access",
+  "RequireNaturallyAlignedBufferAccess",
+  "true",
+  "Requires natural alignment of buffer accesses"
```

Comment:
Should elaborate more. This also seems like the backwards direction to define this. I would assume stricter alignment is the base requirement, and opt-in to relaxed

Reply:
Maybe defining the new feature with 'alignment' in its name is unfortunate. This is really about extra out-of-bounds guarantees that are normally not required.
I will add some more describing text here, similar to what I added in the comment in SIISelLowering.cpp.